### PR TITLE
Waardex Bid Adapter: Add video support, refactor code

### PR DESCRIPTION
**Pull request by yegorWaardex**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Added video support, refactored code 

- test parameters for validating bids
```
{
      bidder: 'waardex',
      params: {
          bidfloor: 1.5,
          position: 1,
          instl: 1,
          zoneId: 1,
          mimes: ['video/x-ms-wmv', 'video/mp4'],
          minduration: 2,
          maxduration: 10,
          protocols: ['VAST 1.0', 'VAST 2.0'],
          startdelay: -1,
          placement: 1,
          skip: 1,
          skipafter: 2,
          minbitrate: 0,
          maxbitrate: 0,
          delivery: [1, 2, 3],
          playbackmethod: [1, 2],
          api: [1, 2, 3, 4, 5, 6],
          linearity: 1,
      }
  }
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Review by FilipStamenkovic — CHANGES_REQUESTED** [review]
Can you update `.md` file to include `zoneId` in your banner examples, even banner request is not valid without `zoneId`?  
Also, I tried testing your `video` changes on my local and my request got rejected with by:
> blocked by CORS policy

How can I test this on my local?  
Also, please open PR on [docs repo](https://github.com/prebid/prebid.github.io) and add support for `video`.

**Comment by yegorWaardex** [comment]
@FilipStamenkovic Hi!

> Can you update .md file to include zoneId in your banner examples, even banner request is not valid without zoneId?

Done here: b0a60b743fd58aac879334120af1c90692554f11

> How can I test this on my local?

Please, go through steps below and if you still have CORS issue then let me know.

1. `npm install`
2. `gulp build`
3. Download [archive](https://drive.google.com/drive/folders/107nMmmRu3PLdySVkc6IUg94jnD9tpUMZ?usp=sharing) with "playground" folder and paste this folder to the root of the project.
4. Open `/playground/video.html` file in the browser

Here is how it looks on my local machine: https://share.getcloudapp.com/04uNXn6X

**Comment by FilipStamenkovic** [comment]
Link to the docs PR: https://github.com/prebid/prebid.github.io/pull/2629

**Review by FilipStamenkovic — APPROVED** [review]
LGTM

**Comment by FilipStamenkovic** [comment]
@yegorWaardex please take a look at this PR: #6200. I had to implement a quick fix, tests were failing for IE 11 because you used `Array.prototype.find` function in this PR.

File: modules/waardexBidAdapter.js [hunk 1]
```diff
@@ -1,217 +1,294 @@
...
+    startdelay: utils.getBidIdParameter('startdelay', videoParams) || 0,
+    placement: utils.getBidIdParameter('placement', videoParams) || videoMediaTypes.context === 'outstream' ? 3 : 1,
+    skip: utils.getBidIdParameter('skip', videoParams) || 1,
+    skipafter: utils.getBidIdParameter('skipafter', videoParams) || 0,
+    minbitrate: utils.getBidIdParameter('minbitrate', videoParams) || 0,
+    maxbitrate: utils.getBidIdParameter('maxbitrate', videoParams) || 3500,
+    delivery: utils.getBidIdParameter('delivery', videoParams) || [2],
+    playbackmethod: utils.getBidIdParameter('playbackmethod', videoParams) || [1, 2, 3, 4],
+    api: utils.getBidIdParameter('api', videoParams) || [2],
+    linearity: utils.getBidIdParameter('linearity', videoParams) || 1
```
Comment (bretg) [inline modules/waardexBidAdapter.js:156]: @yegorWaardex - I know it's too late for this PR, but if I'm reading it correctly, this code breaks the Prebid rule that mediaType video parameters must take precedence over bidder-specific video params.

Our goal is that publishers should only have to specify video parameters in one place -- at the mediatype level. They should not have to copy video params into each bidder's param section.

If I'm misunderstanding the code here, please forgive. If I'm right, please submit another PR to check the video mediatype params first.

thanks

**Comment by patmmccann** [comment]
@yegorWaardex please submit a pr to fix https://github.com/prebid/Prebid.js/pull/6149#discussion_r558585704 and #6512 or your adapter will be removed imminently.

**Pull request merged**